airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py

```diff
@@ -164,9 +164,13 @@ def get_json_schema(self) -> JsonSchema:
         try:
             schema = self._get_raw_json_schema()
         except (InvalidSchemaError, NoFilesMatchingError) as config_exception:
+            self.logger.exception(FileBasedSourceError.SCHEMA_INFERENCE_ERROR.value, exc_info=config_exception)
```

Comment:
Is this always called in the main thread? Else, I fear we might be at risk regarding [this](https://airbytehq-team.slack.com/archives/C063B9A434H/p1705351381223439)